CAIPs/caip-tbd.md

```diff
@@ -0,0 +1,102 @@
+status: Draft
+type: Standard
+created: 2022-10-19
+updated: 2022-10-19
+---
+
+## Abstract
+
+Historically the web platform has had a notion of “powerful” APIs like [geolocation][w3c-geolocation] and [camera and microphone usage][w3c-mediastreams], which are subject to additional security restrictions such as those defined by [secure contexts][w3c-secure-contexts]. Since the Web3 Provider APIs allow dApp websites to request access to sensitive user data and to request use of user funds, new Web3 Provider APIs generally should align to the security considerations of secure context in order to better protect the users data and users funds managed via the web.
+
+## Motivation
+
+Wallets are oftentimes maintaining security and safety of users' funds that can be equivalent to large portions of money. For this reason, it's a good idea to restrict access to the Web3 Provider APIs to align it with other powerful APIs on the web platform. This will assist in reducing the surface area that attacks can be conducted to access users funds or data. Additionally, by adding in restrictions we're reducing the surface area that malicious web pages could fingerprint the user via the Web3 Provider APIs providing some additional privacy benefits. An example of a specific attack that's avoided by this is one where a malicious advertisement is loaded on a legitimate dApp that attempts to interact with a users wallet to maliciously request the user to access funds. With this CAIP implemented the advertisement frame would be considered a third party ("3P") iframe and therefore would not have the Web3 Provider API injected into it's sub frame because it's not a secure context.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Restrictions for providers
+
+The provider objects, e.g. `window.ethereum` or a [Solana wallet adapter][solana-wallet-adapters], are expected to only inject the Web3 provider APIs in secure context when conforming with this standard. The following restrictions are REQUIRED for conformant wallets:
+
+- Provider objects MAY be accessible in private (incognito) windows.
+- The origin MUST be a [potentially trustworthy origin][w3c-secure-context-trustworthy-origin] to have access to the Web3 provider APIs. This can be checked using `window.isSecureContext`, including inside iframes.
+    - Secure contexts include sites that are served from HTTPS but also HTTP `localhost`. 
+    - The browser implementation may optionally also support configured [potentially trustworthy origins] that would normally not be considered trustworthy if the user configures their browser to do so. See the [Development Environments section of Secure Contexts][w3c-secure-context-dev-env] for additional details.
+- By default the Web3 Provider APIs MUST NOT be exposed to 3p iframes.
```

Comment:
```suggestion
- By default the Web3 Provider APIs MUST NOT be exposed to third-party iframes.
```

Is this semantically equivalent? Regardless, what does '3p' or even 'third-party' mean exactly? (i.e. different origin? different apex domain?)

Reply:
It's probably a good idea to align this with other cross-origin restrictions.

Reply:
yup those are equivalent @gobengo.

> It's probably a good idea to align this with other cross-origin restrictions.
@darobin yeah that's my thinking here is that this is really focused on when we're dealing with different origins. I was under the impression that third party iframes would imply that meant different origins, but the fact that it's implied makes for imprecise language typically. Do you have suggestions about how to better word this?